Binary files filter-spamassassin-0.7/filter-spamassassin and filter-spamassassin-0.7-new/filter-spamassassin differ
diff -rupN filter-spamassassin-0.7/filter-spamassassin.go filter-spamassassin-0.7-new/filter-spamassassin.go
--- filter-spamassassin-0.7/filter-spamassassin.go	2020-04-18 01:05:13.000000000 +0200
+++ filter-spamassassin-0.7-new/filter-spamassassin.go	2024-01-17 17:01:59.849266492 +0100
@@ -121,7 +121,8 @@ func (sa *spamassassin) response(tok str
 }
 
 func (sa *spamassassin) process(tok string) {
-	con, e := net.Dial("tcp", addr)
+	var tcpsocket bool = strings.Contains(addr, ":")
+	con, e := net.Dial(map[bool]string{true: "tcp", false: "unix"}[tcpsocket], addr)
 	if e != nil {
 		l3.Err(fmt.Sprintln(sa.sid, e))
 		sa.reset(tok)
@@ -140,7 +141,13 @@ func (sa *spamassassin) process(tok stri
 			return
 		}
 	}
-	if c, ok := con.(*net.TCPConn); ok {
+	if (tcpsocket) {
+		c, _ := con.(*net.TCPConn)
+		if e = c.CloseWrite(); e != nil {
+			l3.Warning(fmt.Sprintln(sa.sid, "closewrite", e))
+		}
+	} else {
+		c, _ := con.(*net.UnixConn)
 		if e = c.CloseWrite(); e != nil {
 			l3.Warning(fmt.Sprintln(sa.sid, "closewrite", e))
 		}
@@ -211,7 +218,7 @@ func run() {
 	for in.Scan() {
 		f := strings.Split(in.Text(), "|")
 		t, ver, ev, sid := f[0], f[1], f[4], f[5]
-		if (t != "filter" && t != "report") || ver != "0.6" {
+		if (t != "filter" && t != "report") || ver != "0.7" {
 			l3.Err(fmt.Sprintln(sid, "protocol", t, ver))
 			return
 		}
